kv/dist_sender.go

```diff
@@ -544,6 +597,26 @@ func (ds *DistSender) Send(call *client.Call) {
 	}
 }
 
+// updateLeaderCache updates the cached leader for the given Raft group,
+// evicting any previous value in the process.
+// The new leader is cached only if it isn't equal to the newly evicted value.
+func (ds *DistSender) updateLeaderCache(rid proto.RaftID, leader proto.Replica) {
+	// Slight overhead here since we want to make sure that when the new
+	// proposed leader equals the old, we only evict.
```

Comment:
Could you expand this comment to explain that we evict in the event that the leader is the same as the previously cached leader because we might be stuck repeating calls to the same bad replica and need to try one at random.